decidim-admin/app/commands/decidim/admin/reorder_participatory_process_steps.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+module Decidim
+  module Admin
+    # A command that sets a step in a participatory process as active (and
+    # unsets a previous active step)
+    class ReorderParticipatoryProcessSteps < Rectify::Command
+      # Public: Initializes the command.
+      #
+      # collection - an ActiveRecord::Relation of steps
+      # order_string - a String representing an Array holding the order of IDs
+      #   of steps
+      def initialize(collection, order_string)
+        @collection = collection
+        @order_string = order_string
+      end
+
+      # Executes the command. Braodcasts these events:
+      #
+      # - :ok when everything is valid.
+      # - :invalid if the data wasn't valid and we couldn't proceed.
+      #
+      # Returns nothing.
+      def call
+        return broadcast(:invalid) unless order_string.present?
+        begin
+          @order = JSON.parse(order_string)
+        rescue JSON::ParserError
+          return broadcast(:invalid)
+        end
+        return broadcast(:invalid) unless order.is_a?(Array) && order.present?
+
+        reorder_steps
+        broadcast(:ok)
+      end
+
+      private
+
+      attr_reader :order_string, :collection, :order
+
+      def reorder_steps
+        data = {}
+        order.each_with_index do |id, index|
+          data[id] = { position: index }
+        end
+
+        collection.update(data.keys, data.values)
```

Comment:
Your version does an update for every single element in the collection. My version uses a single update for all the collection: 

https://github.com/rails/rails/blob/a419a4d9ade48e777166ff956dd7bb24e37b2181/activerecord/lib/active_record/relation.rb#L363